package com.basicsOOPs;
public class ExceptionHandlingExample {
	public static void m1()
	{
		String j="java";
		try {
			int i=Integer.parseInt(j);
			System.out.println(i);
			
		} catch (NumberFormatException e) {
			System.out.println("handled");
		}
		finally
		{
			System.out.println("final method in java");
		}
	}
	public static void checkedExceptions()
	{
		System.out.println("starting point");
		try {
			Thread.sleep(2000);
		} catch (InterruptedException e) {
			System.out.println("handled....");
		}
	}
	public static void main(String[] args) {
//	Null pointer excweption
		String s=null;
//		System.out.println("null pointer exception example");
//		try {
//			System.out.println(s.length());
//		} catch (NullPointerException e) {
//			System.out.println("handled");
//		}
		int a=12;int b=0;
//		try {
//			System.out.println(a/b);
//			
//		} catch (ArithmeticException e) {
//			System.out.println("handled");
//			b=20;
//			System.out.println(b);	
//		}
		
		
		  
//		String m="IU";
//		System.out.println(m);
//		try {
//			int j=Integer.parseInt(m);
//			System.out.println(j*10);
//			
//		} catch (Exception e) {
//			System.out.println("Handled");
//		}
//		int[] a1= {32,432,32,342,34,23};
//		try {
//			System.out.println(a1[10]);
//			
//		} catch (ArrayIndexOutOfBoundsException e) {
//			System.out.println("handled");
//			System.out.println(e.getMessage());
//			e.printStackTrace();
//		}
//		//multiple try catch 
//		
//		try {
//			System.out.println(a/b);
//			
//		} catch (ArithmeticException e) {
//			System.out.println("Handled");
//		}catch(ArrayIndexOutOfBoundsException e)
//		{
//			System.out.println("handled for array index bound of exception ");
//		}
		
		
		//m1();
		checkedExceptions();
	}
}
